libraries/dictionary/src/types/dictionaryTypes.ts

```diff
@@ -29,11 +29,20 @@ export type NameString = zod.infer<typeof NameString>;
 
 export const Integer = zod.number().int();
 
-// Unlike references, meta is not nested and accepts as values only primitives or arrays of them.
-export const DictionaryMetaValue = zod.union([zod.string(), zod.number(), zod.boolean()]);
+// Meta accepts as values only stirngs, numbers, booleans, arrays of numbers or arrays of strings
+// Another Meta object can be nested inside a Meta property
+export const DictionaryMetaValue = zod.union([
+	zod.string(),
+	zod.number(),
+	zod.boolean(),
+	zod.array(zod.string()),
+	zod.array(zod.number()),
+]);
 export type DictionaryMetaValue = zod.infer<typeof DictionaryMetaValue>;
```

Comment:
this line reads confusingly, due to the identical capital capitalisation of type and variable to infer from.

maybe naming itself should be different? (e.g. additional description for each to differentiate one from the other)

Reply:
This is a common recurring pattern with type schemas. We need to either rename all the schema objects to have names with the Schema suffix or accept that there are types and schemas with the same name. The latter is proposed in the zod documentation and is the pattern followed here.

Reply:
in other words, limit the standards for the sake of a library... just like `import z` opinion?

questionable

Reply:
Difference here is I like this pattern. The schema validates for a type of the same name, it's quite intuitive to work with.

Reply:
that may be so for those without cognitive impairments, perhaps. 🤷‍♂️

PR was already approved anyway, declaritive enough, or not. ✅